sphere-java-client-core/src/main/java/io/sphere/sdk/client/BlockingSphereClientImpl.java

```diff
@@ -0,0 +1,46 @@
+package io.sphere.sdk.client;
+
+import java.util.concurrent.*;
+
+ class BlockingSphereClientImpl implements BlockingSphereClient {
+
+    private final SphereClient delegate;
+    private final long defaultTimeout;
+    private final TimeUnit unit;
+
+    public BlockingSphereClientImpl(final SphereClient delegate, final long defaultTimeout, final TimeUnit unit) {
+        this.delegate = delegate;
+        this.defaultTimeout = defaultTimeout;
+        this.unit = unit;
+    }
+
+    @Override
+    public void close() {
+        delegate.close();
+    }
+
+    @Override
+    public <T> CompletionStage<T> execute(final SphereRequest<T> sphereRequest) {
+        return delegate.execute(sphereRequest);
+    }
+
+    @Override
+    public <T> T executeBlocking(final SphereRequest<T> sphereRequest) {
+        return executeBlocking(sphereRequest, defaultTimeout, unit);
```

Comment:
a pitfall with that approach is that the timeout on the future can throw an exception before the timeout of the http client.
It means:
- the user receives a timeout
- but the request eventually succeed.

We have to be careful here to use a timeout long enough (more than the request timeout of the underlying http client)